packages/battery/example/android/app/src/main/java/io/flutter/plugins/batteryexample/EmbedderV1Activity.java

```diff
@@ -6,12 +6,14 @@
 
 import android.os.Bundle;
 import io.flutter.app.FlutterActivity;
-import io.flutter.plugins.GeneratedPluginRegistrant;
+import io.flutter.plugins.battery.BatteryPlugin;
+import dev.flutter.plugins.e2e.E2EPlugin;
 
 public class EmbedderV1Activity extends FlutterActivity {
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
-    GeneratedPluginRegistrant.registerWith(this);
+    BatteryPlugin.registerWith(registrarFor("io.flutter.plugins.battery.BatteryPlugin"));
```

Comment:
This part is a bit unfortunate but we built infra to generate for v1 embedding or v2 embedding toggling between v1 and v2 plugins. We never build infra for toggling between v1 and v2 embedding. 

This is probably ok since it's mostly transient but we have to document it in https://github.com/flutter/website/pull/3388